pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +73,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := eventClient.Watch(
```

Comment:
> actually, doesn't this want to pick up events that already exist? I think watch with no resourceVersion is what is desired

Seems like that would make its usage below wrong nearly 100% of the time, right?  The `ImageStream` almost certainly pre-dates this particular push.